WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```
@@ -229,10 +233,53 @@ private extension PaymentCaptureOrchestrator {
                                  currency: order.currency,
                                  receiptDescription: receiptDescription(orderNumber: order.number),
                                  statementDescription: statementDescriptor,
+                                 receiptEmail: provideReceiptEmailIfNecessary(from: order),
                                  paymentMethodTypes: paymentMethodTypes,
                                  metadata: metadata)
     }
 
+    private func synchronizePlugins(from siteID: Int64) async throws {
+        return try await withCheckedThrowingContinuation({
+                (continuation: CheckedContinuation<(), Error>) in
+            let systemPluginsAction = SystemStatusAction.synchronizeSystemPlugins(siteID: siteID) { result in
+                if case let .failure(error) = result {
+                    DDLogError("[PaymentCaptureOrchestrator] Error syncing system plugins: \(error)")
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume(returning: ())
+                }
+            }
+            ServiceLocator.stores.dispatch(systemPluginsAction)
+            })
+    }
+
+    /// We do not need to set the receipt email if WCPay is installed and active
+    /// and its version is higher or equal than 4.0.0, as it does itself in that case.
+    private func provideReceiptEmailIfNecessary(from order: Order) -> String? {
+        let paymentsPluginsDataProvider = PaymentsPluginsDataProvider()
+
+        let wcPay = paymentsPluginsDataProvider.getWCPayPlugin()
+        let stripe = paymentsPluginsDataProvider.getStripePlugin()
+
+        guard !paymentsPluginsDataProvider.bothPluginsInstalledAndActive(wcPay: wcPay, stripe: stripe) else {
+            // This case should not happen, shall we fatal error here?
```

My 1ct - yes better to fatal, because this means that something really wrong goes in the code and the feature won't work anyway

---

A silent error is not ideal. I do not love the idea of crashing on purpose, but at the very least, we should log the reason why we stop execution, and we should try to inform the user of what is going on. 

In this particular case, we have seen this case happening in production. There have been users with both WCPay and the Stripe plugin installed, so we should guard against it properly.

---

> In this particular case, we have seen this case happening in production. There have been users with both WCPay and the Stripe plugin installed, so we should guard against it properly.

Please correct me if on iOS it's not the case. We should not build `PaymentIntent` when IPP is not available for the site, but instead, we should show the proper error screen that I think is already in place for iOS too and user won't be able to start collecting payment if 2 plugins are active?

If this is correct, then returning null from here will end up in the error/nonfuncinalal payment somewhere later and we won't probably see a clear picture of why it happens

**Fatal approach**
* Quick feedback loop from sentry
* Crash stack clear says what and where in the code it happened 

**Logging and returning null**
* People don't check logs mostly so the issue will be in place longer
* Even if users will start complaining that IPP doesn't work in some cases it will be not easy to figure out why

The result is the same - IPP doesn't work in both approaches, but with the first approach we will fix it faster, spending less time on investigation

---

We still have the issue with Order Details: pdfdoF-yu-p2

It is true that we should not start creating a PaymentIntent if the store and the order is not eligible, but the fact is that, due to an oversight on our side, it is possible. 

Crashing when the merchant is trying to capture a payment does not sound like the best way to communicate that something is not quite right. Showing an error message and aborting the attempt to capture a payment sounds to me like a better way to fail.

And if we log the error we can still look at the application logs to trace exactly what happened.

---

😅 I knew this would be a spicy one. It was my suggestion on our sync call yesterday that @toupper consider crashing here. 

I'm not stirring up trouble deliberately, I'm really not sure which way we should go for all the above reasons.

Probably in this case, crashing because they have both installed is overzealous – all we're using the information for is to set an optional email parameter. So returning `nil` is not so bad, the things which people care about most (the payment working and being correct) will happen even if we set the email to nil incorrectly.

If we want to crash for having both plugins installed after the connection to the reader has happened, perhaps that would be better done (in other work) at the point where we decide which plugin to use...

---

Thank you all for the feedback 🎉 ! As it seems we have these options:

1. Fatal error and crash: Pros: Quick and meaningful feedback. Cons: Not a good user experience especially in a critical user case (payment).
2. Payment Error:
           - Silently: Pros: Less code and complexity to implement Cons: Not a good experience for the user, who might not know what is happening.
           - With a UI Message: Pros: Good user experience, communicate to the user what is happening. Cons: Payment fails (is it necessary?)
3. Go with the payment but return nil receipt email: Pros: Less code and complexity, the payment goes on. Cons: Would the payment work if both of them are installed and active? If that is the case, I would go for this one for this hotfix and create an issue to properly handle this case on the payment in the future.

---

Moved discussion here pdfdoF-CS-p2

---

The outcome of the discussion:
- For the receipt email case, we do not send it if we have WCPay and Stripe both installed and active at the same time. We do not fail or crash.
- We implement the [onboarding checks](https://href.li/?https://github.com/woocommerce/woocommerce-ios/issues/6607) in another PR